addons/mail/static/tests/helpers/mock_server/models/mail_activity.js

```diff
@@ -118,57 +126,167 @@ patch(MockServer.prototype, {
      * @param {number} offset
      * @returns {Object}
      */
-    _mockMailActivityGetActivityData(res_model, domain, limit = 0, offset = 0) {
+    /**
+     * Simulate partially (time zone not supported) `_compute_state_from_date` on `mail.activity`.
+     *
+     * @param {DateTime} date_deadline to convert into state
+     * @returns {string} dateline status (today, planned or overdue)
+     * @private
+     */
+    _mockComputeStateFromDate(date_deadline) {
+        const now = DateTime.now();
+        if (date_deadline.hasSame(now, "day")) {
+            return "today";
+        } else if (date_deadline > now) {
+            return "planned";
+        }
+        return "overdue";
+    },
+    /**
+     * Simulates `get_activity_data` on `mail.activity`.
+     *
+     * @private
+     * @param {string} res_model
+     * @param {string} domain
+     * @param {number} limit
+     * @param {number} offset
+     * @param {boolean} fetch_done
+     * @returns {Object}
+     */
+    _mockMailActivityGetActivityData(res_model, domain, limit = 0, offset = 0, fetch_done = false) {
         const self = this;
-        const allRecords = this.getRecords(res_model, domain);
+
+        // 1. Retrieve all ongoing and completed activities according to the parameters
+        const activityTypes = this.getRecords("mail.activity.type", [
+            "|",
+            ["res_model", "=", res_model],
+            ["res_model", "=", false],
+        ]);
+        // Remove domain term used to filter record having "done" activities (not understood by the getRecords mock)
+        domain = Domain.removeDomainLeaves(new Domain(domain ?? []).toList(), [
+            "activity_ids.active",
+        ]).toList();
+        const allRecords = this.getRecords(res_model, domain ?? []);
         const records = limit ? allRecords.slice(offset, offset + limit) : allRecords;
+        const activityDomain = [["res_model", "=", res_model]];
```

Comment:
Globally, same naming proposals apply here compared to python, if possible to quickly go through and simplify a bit :) (like allOngoingActivities -> ongoing, groupedCompletedActivities -> groupCompleted, ...)